lib/components/molecules/DateSelector/DateSelector.js

```diff
@@ -0,0 +1,203 @@
+import Label from '../../atoms/Label';
+
+const monthOptions = [
+  'January',
+  'Febuary',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+];
+
+const DateSelector = (props: Props): Node => {
+  const { className, format, startDate, endDate } = props;
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const [newStartDate, setNewStartDate] = useState(
+    new Date(new Date().setFullYear(new Date().getFullYear() - 100))
+  );
+  const [newEndDate, setNewEndDate] = useState(
+    new Date(new Date().setFullYear(new Date().getFullYear() + 50))
+  );
+
+  const setDate = dateType => {
+    if (dateType === 'start') {
+      if (startDate) {
+        setNewStartDate(new Date(startDate));
+      } else {
+        const d = new Date();
+        setNewStartDate(new Date(d.setFullYear(d.getFullYear() - 100)));
+      }
+    } else if (dateType === 'end') {
+      if (endDate) {
+        setNewEndDate(new Date(endDate));
+      } else {
+        const d = new Date();
+        setNewEndDate(new Date(d.setFullYear(d.getFullYear() + 50)));
+      }
+    }
+  };
+
+  useEffect(() => {
+    // ... Use hooks
+    setDate('start');
+    setDate('end');
+  }, []);
+
+  useEffect(() => {
+    setSelectedDate(newStartDate);
+  }, [newStartDate, newEndDate]);
+
+  const getDaysInMonth = () => {
+    const noOfDays = new Date(selectedDate.getFullYear(), selectedDate.getMonth(), 0).getDate();
+    const days = [];
+    let startDay = 1;
+    let endDay = noOfDays;
+    if (
+      selectedDate.getMonth() === newStartDate.getMonth() &&
+      selectedDate.getFullYear() === newStartDate.getFullYear()
+    ) {
+      startDay = newStartDate.getDate();
+    }
+    if (
+      selectedDate.getMonth() === newEndDate.getMonth() &&
+      selectedDate.getFullYear() === newEndDate.getFullYear()
+    ) {
+      endDay = newEndDate.getDate();
+    }
+    for (let i = startDay; i <= endDay; i += 1) {
+      days.push(i);
+    }
+    return days;
+  };
+
+  const getMonthOptions = () => {
+    let months = [...monthOptions];
+    if (selectedDate.getFullYear() === newStartDate.getFullYear()) {
+      const startMonth = newStartDate.getMonth();
+      months = months.slice(startMonth, months.length);
+    }
+    if (selectedDate.getFullYear() === newEndDate.getFullYear()) {
+      const endMonth = newEndDate.getMonth();
+      months = months.slice(0, endMonth + 1);
+    }
+    return months;
+  };
+
+  const getYearOptions = () => {
+    const years = [];
+    for (let i = newStartDate.getFullYear(); i <= newEndDate.getFullYear(); i += 1) {
+      years.push(i);
+    }
+    return years;
+  };
+
+  const DaySelector = (
+    <Label htmlFor="day">
+      <Select
+        onChange={e => {
+          return setSelectedDate(
+            new Date(selectedDate.getFullYear(), selectedDate.getMonth(), e.target.value)
+          );
+        }}
+        id="day"
```

Comment:
Don't hard code id, create one from the one passed for whole component

Reply:
Updated